tonic/models/vic/vic.py

```diff
@@ -95,11 +101,28 @@ def _call_vic(self, *args, **kwargs):
 
         vic_args = []
 
-        if "mpi_proc" in kwargs.keys():
-            if not isinstance(kwargs["mpi_proc"], int):
-                raise TypeError("number of processors must be specified as an integer")   
- 
-            vic_args.extend(['mpirun', '-np', '%.0d' % kwargs["mpi_proc"]])  
+        # Get mpi info
+        mpi_proc = kwargs.pop('mpi_proc', None)
+        if mpi_proc is not None:
+            if not isinstance(mpi_proc, int):
+                raise TypeError("number of processors must be specified as an"
+                                "integer")
+            vic_args.extend(['mpirun', '-np', '%.0d' % kwargs["mpi_proc"]])
+
+        # Get valgrind info
+        valgrind = kwargs.pop('valgrind', None)
+        if valgrind is not None:
+            if valgrind is True:
+                valgrind = 'valgrind'
+            errorcode = os.getenv('VIC_VALGRIND_ERROR',
+                                  default_vic_valgrind_error_code)
+            vic_args.extend([valgrind, '-v', '--leak-check=full',
+                             '--error-exitcode={0}'.format(errorcode)])
+
+        # if there are kwargs left, we don't know what to do with them so
+        # raise an error
+        if kwargs:
+            raise ValueError('Unknown argument(s): %s' % ', '.join(kwargs.keys()))
```

Comment:
Absolutely, I should have caught that you didn't add this line to your PR.  Imagine the case where you set `mpiproc=4` (note the missing underscore).  VIC would run and you would never know you specified a parameter that wasn't used.  This is defensive programing.